cmake/toolchain/llvm/generic.cmake

```diff
@@ -8,8 +8,8 @@ if((NOT "${USER_CACHE_DIR}" STREQUAL "") AND (EXISTS "${USER_CACHE_DIR}"))
     ${CMAKE_COMMAND} -E remove_directory "${USER_CACHE_DIR}")
 endif()
 
-if(DEFINED $ENV{CLANG_ROOT_DIR})
-  set(TOOLCHAIN_HOME ${CLANG_ROOT}/bin/)
+if(DEFINED ENV{CLANG_ROOT_DIR})
+  set(TOOLCHAIN_HOME $ENV{CLANG_ROOT_DIR}/bin/)
```

Comment:
Thanks for this, but could you update to use: `LLVM_TOOLCHAIN_PATH`, so that if follows same name form as other toolchains (and I believe `/bin` is superfluous)

See also: https://github.com/zephyrproject-rtos/zephyr/pull/23120#issuecomment-646555798